tel2puml/find_unique_graphs/otel_ingestion/otel_data_holder.py

```diff
@@ -2,19 +2,171 @@
 
 from abc import ABC, abstractmethod
 
+from sqlalchemy import create_engine, insert
+from sqlalchemy.orm import Session
+from sqlalchemy.engine.base import Engine
+
 from tel2puml.find_unique_graphs.otel_ingestion.otel_data_model import (
     OTelEvent,
+    SQLDataHolderConfig,
+    NodeModel,
+    Base,
+    NODE_ASSOCIATION,
 )
 
 
 class DataHolder(ABC):
     """An abstract class to handle saving processed OTel data."""
 
-    @abstractmethod
+    def __init__(self) -> None:
+        """Constructor method."""
+
+        self.min_timestamp: int | None = None
```

Comment:
I would initialise these to 0 and the largest conceivable integer, respectively. Then you won't have to do if statements below